src/api/db/data/20200421121610_backfill_notified_projects.rb

```diff
@@ -0,0 +1,25 @@
+class BackfillNotifiedProjects < ActiveRecord::Migration[6.0]
+  def up
+    Notification.where.not(notifiable_id: nil, notifiable_type: nil).find_each do |notification|
+      case notification.notifiable_type
+      when 'BsRequest'
+        notification.projects << notification.notifiable.target_project_objects.uniq
+      when 'Comment'
+        case notification.notifiable.commentable_type
+        when 'Project'
+          notification.projects << notification.notifiable.commentable
+        when 'Package'
+          notification.projects << notification.notifiable.commentable.project
+        when 'BsRequest'
+          notification.projects << notification.notifiable.commentable.target_project_objects.uniq
+        end
+      when 'Review'
+        notification.projects << notification.notifiable.bs_request.target_project_objects.uniq
+      end
+    end
+  end
```

Comment:
:thinking: We could have problems if we run this data migration without deploying the changes that associate notifications and projects (right after notification's creation).

The reason is that since the data migration is run and until these changes are deployed,  new notifications will be created but won't associate projects to them. 

If we do this the other way around (deploying the code first and running this migration afterwards) the problem will be that we'll try to insert duplicated data. Maybe this could be solved by emptying `notified_projects` table as first step in this data migration.

We can talk about this offline in case my explanation is not clear and decide what to do afterwards.

Reply:
We could always rerun the data migration, it won't create duplicate records anyway. However, we need to rescue from `ActiveRecord::RecordNotUnique`

Reply:
As we discussed offline, possible duplication exceptions are going to be handled properly. 

Running the migration twice is not possible as it is tied to versions (in db/data_schema.rb), therefore the `down` part is going to be prepared to empty the whole table in case we need to rollback and migrate again (what we'll try to avoid).